plonky2/src/gadgets/lookup.rs

```diff
@@ -89,25 +89,53 @@ impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
 
                 let lookups = self.get_lut_lookups(lut_index).to_owned();
 
-                for (looking_in, looking_out) in lookups {
-                    let gate = LookupGate::new_from_table(&self.config, lut.clone());
+                let gate = LookupGate::new_from_table(&self.config, lut.clone());
+                let num_slots = LookupGate::num_slots(&self.config);
+
+                /* Given the number of lookups and the number of slots for each gate, it is possible
+                 to compute the number of gates that will employ all their slots; for such gates,
+                 the `num_slots` calls to the `find_slot` function can be replaced with a single
+                 `add_gate` call.
+                */
```

Comment:
nit: We don't really use `/* */` for comments elsewhere and usually use `//` instead.